keps/sig-auth/3221-structured-authorization-configuration/README.md

```diff
@@ -175,11 +175,12 @@ apiVersion: apiserver.config.k8s.io/v1alpha1
 kind: AuthorizationConfiguration
 authorizers:
   - type: Webhook
-    name: system-crd-protector
```

Comment:
huh, I missed that the design had the name at the authorizer entry level and the implementation had the name at the webhook config level.

I think we probably want a unique name to exist at the authorizer entry level as shown here, so that metrics / audit / logging can consistently use the name of the authorizer, regardless of the type.

We would probably make the name for built-in authz modes just be lower(mode), so `rbac`, `node`, `abac`, etc, and validate that in kube-apiserver specifically for the modes it knows about.

Reply:
> We would probably make the name for built-in authz modes just be lower(mode), so rbac, node, abac, etc, and validate that in kube-apiserver specifically for the modes it knows about.

Wouldn't adding a new mode in the future be backwards incompatible then since it could conflict with the name of a webhook (and names need to be unique)?

Reply:
Maybe not because you always have to opt-in to what authorizers you want to use?

Reply:
yeah, not a problem, if you want to use a new built-in one, then you need to deconflict your names if we happen to have one that matches a name you chose for one of your webhooks

Reply:
@palnabarun restore the name to the top-level type, and drop it from the webhook stanza, and we'll get a PR in place to make the code match